walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -10,14 +10,55 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License.
+// limitations under the Liceunse.
 
 package com.google.sps;
 
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Iterator;
+
 
 public final class FindMeetingQuery {
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    Collection<String> attendees = request.getAttendees();
+    long duration = request.getDuration();
+
+    // no attendees case, Expected :[Range: [0, 1440)]
+    if(attendees.isEmpty()){
+
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+
+    }
+    // during over 24hours , Expected :[Range: [0, 1440)]
+    if (duration> TimeRange.WHOLE_DAY.duration()){
+      return Arrays.asList();
+    }
+
+    Iterator<Event> it = events.iterator();
```

Comment:
How will we make sure the attendees in the meetingRequest are those in the events?